docs/site/migration/models/remoting-hooks.md

```diff
@@ -6,8 +6,266 @@ sidebar: lb4_sidebar
+enables you to execute a function before or after a remote method is called by a
+client:
+
+- `beforeRemote()` runs before the remote method.
+- `afterRemote()` runs after the remote method has finished successfully.
+- `afterRemoteError()` runs after the remote method has finished with an error.
+
+Hooks can be configured to run for one or more methods using a wildcard
+specifier. In general, there are three kinds of hook scopes:
+
+- **Global hooks** are executed for _every_ remote method.
+
+  These hooks are registered on the application object using a `**` wildcard.
+  For example:
+
+  ```js
+  app.beforeRemote('**', handlerFn);
+  ```
+
+- **Model-level hooks** are executed for every remote method of a given model
+  class.
+
+  These hooks are typically registered on the application object using
+  `{model-name}.**` wildcard. For example, to run a hook for all methods of the
+  `User` model:
+
+  ```js
+  app.beforeRemote('User.**', handlerFn);
+  ```
+
+- **Method-level hooks** are executed only for a single method of a given model
+  class.
+
+  These hooks are typically registered on the model class. For example, to run a
+  hook whenever `User.login` is called remotely:
+
+  ```js
+  User.beforeRemote('User.login', handlerFn);
+  ```
+
+LoopBack 4 provides [Interceptors](../../Interceptors.md) feature to enable
+application developers to implement similar functionality. In the following
+sections, we will first explain how to rewrite hook implementations to
+interceptors and then show how to register interceptors to be invoked globally,
+at a class level or at a method level.
+
+## Migrating a hook to an interceptor
+
+Let's take a look at a typical interceptor implementation:
+
+```ts
+async function intercept(
+  invocationCtx: InvocationContext,
+  next: () => ValueOrPromise<InvocationResult>,
+) {
+  try {
+    // Add pre-invocation logic here
+    const result = await next();
+    // Add post-invocation logic here
+    return result;
+  } catch (err) {
+    // Add error handling logic here
+    throw err;
+  }
+}
+```
+
+Every interceptor contains three parts, each corresponding to one of the
+remoting hooks:
+
+- Code from a `beforeRemote` hook belongs to the first section that's executed
+  before the target method is invoked.
+- Code from an `afterRemote` hook belongs to the second section.
+- Code from an `afterRemoteError` hook belongs to the third section inside the
+  `catch` block.
+
+{% include tip.html content=" In LoopBack 3, hooks can be written in one of the
+following three styles:
+
+- A function accepting a callback as the last argument.
+- A function returning a promise, often using
+  [Bluebird](http://bluebirdjs.com/docs/getting-started.html) APIs.
+- An async function using `await` statements for asynchronous flow control.
+
+In LoopBack 4, interceptors are written as async functions using `await`
+statements. "%}
+
+### Example
+
+Consider the following set of remote hooks implementing a very simple request
```

Comment:
Same here: `Consider the following set of LB3 remote hooks implementing a very simple request`..